File: 프로그래머스/3/42884. 단속카메라/단속카메라.java
import java.util.*;

class Solution {
    public int solution(int[][] routes) {
        Arrays.sort(routes, (o1, o2) -> Integer.compare(o1[1], o2[1]));
        
        int answer = 1;
        int last = routes[0][1];
        
        for(int i = 1; i < routes.length; i++){
            if(last < routes[i][0]){
                last = routes[i][1];
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/3/42884. 단속카메라/단속카메라.js
function solution(routes) {
    routes.sort((a, b) => (a[1] - b[1]));
    
    let ans = 1;
    let camera = routes[0][1];
    
    for(let i = 1; i < routes.length; i++){
        if(camera < routes[i][0]){
            camera = routes[i][1];
            ans++;
        }
    }
    return ans;
}